### PR TITLE
fix: jssStyle type

### PR DESCRIPTION
**Pull request by hosseinmd**
typescript can't work with   [K in keyof NormalCssProperties | string]

## What would you like to add/fix?
type definition
## Todo

- [ ] Add tests if possible
- [ ] Add changelog if users should know about the change
- [ ] Add documentation

### CODE REVIEW

**Comment by kof** [comment]
What is the actual error compiler was reporting? This fix seems fishy, need some ts experts to confirm cc @moshest

**Comment by hosseinmd** [comment]
the bug is where intellisense no suggestions can't suggest CSS Properties.

**Comment by hosseinmd** [comment]
Because we tell them k is CSS properties or string, which means any string.

**Comment by kof** [comment]
oh I see, that makes sense

**Comment by moshest** [comment]
Why we remove there the `Observable<NormalCssValues<K> | JssStyle | undefined>` part?

**Comment by hosseinmd** [comment]
> Why we remove there the `Observable<NormalCssValues<K> | JssStyle | undefined>` part?

Hi, You are right, that is my mistake, I fix that.

File: packages/jss/src/index.d.ts [hunk 1]
```diff
@@ -15,13 +15,17 @@ type NormalCssValues<K> = K extends keyof NormalCssProperties
...
+export type JssStyle =
+  | {
+      [K in keyof NormalCssProperties]:
+        | NormalCssValues<K>
+        | JssStyle
+        | Func<NormalCssValues<K> | JssStyle | undefined>
+        | Observable<NormalCssValues<K> | JssStyle | undefined>
+    }
+  | {
+      [K: string]: JssStyle | Func<JssStyle> | Observable<JssStyle | undefined>
```
Comment (kof) [inline packages/jss/src/index.d.ts:27]: why is this value different than the first value, isn't it supposed to be the exact saame thing?

```
        | NormalCssValues<K>
        | JssStyle
        | Func<NormalCssValues<K> | JssStyle | undefined>
        | Observable<NormalCssValues<K> | JssStyle | undefined>
```

Comment (hosseinmd) [reply packages/jss/src/index.d.ts:27]: Actually no, because the string is not a key of NormalCssValues.

Comment (hosseinmd) [reply packages/jss/src/index.d.ts:27]: I just removed NormalCssValues

Comment (hosseinmd) [reply packages/jss/src/index.d.ts:27]: NormalCssValues has no effect there

Comment (hosseinmd) [reply packages/jss/src/index.d.ts]: But I missed jssValue

**Comment by hosseinmd** [comment]
@kof please review the last commit

**Comment by moshest** [comment]
lgtm

**Comment by kof** [comment]
merged, thanks

**Comment by hosseinmd** [comment]
Could you please publish master?

**Comment by hosseinmd** [comment]
Please publish

**Comment by kof** [comment]
Released https://github.com/cssinjs/jss/releases/tag/v10.5.1

**Pull request merged**